core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -92,12 +92,17 @@ sealed abstract class UnsupportedMediaTypeFailure(expected: Set[MediaRange]) ext
 final case class MediaTypeMissing(expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  val message = s"Decoder is unable to decode a Message without a MediaType. Expected media ranges: $expected"
+  val expectedMsg = expected.map(_.renderString).mkString(", ")
+  val message = s"No media type specified in Content-Type header. Expected one of the following media ranges: $expectedMsg"
 }
 
 /** Indicates that no [[EntityDecoder]] matches the [[MediaType]] of the [[Message]] being decoded */
 final case class MediaTypeMismatch(messageType: MediaType, expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  def message = s"$messageType is not a supported media type. Decoder accepts one of the following media ranges: $expected"
+  def message = {
+    val actual = messageType.renderString
+    val expectedMsg = expected.map(_.renderString).mkString(", ")
+    s"$actual is not a supported media type. Expected one of the following media ranges: $expectedMsg"
```

Comment:
That's an interesting question on the intersection of valid and malicious. I suppose it would need one slash.

Mime sniffing is something that has to be explicitly disabled by proprietary header in certain browsers.  It's ridiculous.  I can't think of any current browser that wouldn't be safe with that additional header, but I do enjoy our record of not having issues with "CVE" in the title. :)